x-pack/plugins/monitoring/server/kibana_monitoring/collectors/get_ops_stats_collector.js

```diff
@@ -6,47 +6,48 @@
 
 import { KIBANA_STATS_TYPE } from '../../../common/constants';
 import { opsBuffer } from './ops_buffer';
-import { Collector } from '../classes/collector';
+import { Collector } from '../classes';
+
+let monitor = null; // singleton because start and stop can be called repeatedly because of status event repetitions
 
 /*
  * Initialize a collector for Kibana Ops Stats
- * FIXME: https://github.com/elastic/x-pack-kibana/issues/1301
  */
 export function getOpsStatsCollector(server) {
-  let monitor;
   const buffer = opsBuffer(server);
   const onOps = event => buffer.push(event);
-  const init = () => {
+
+  const start = () => {
     monitor = server.plugins['even-better'].monitor;
     monitor.on('ops', onOps);
   };
-
-  const cleanup = () => {
+  const stop = () => {
     if (monitor) {
       monitor.removeListener('ops', onOps);
     }
+    monitor = null;
   };
 
-  // This needs to be removed once the FIXME for 1301 is fixed
+  /* Handle stopping / restarting the event listener if Elasticsearch stops and restarts
+   * NOTE it is possible for the plugin status to go from red to red and
+   * trigger handlers twice
+   */
+  server.plugins.elasticsearch.status.on('red', stop);
+  server.plugins.elasticsearch.status.on('green', start);
+
   // `process` is a NodeJS global, and is always available without using require/import
   process.on('SIGHUP', () => {
     this.log.info('Re-initializing Kibana Monitoring due to SIGHUP');
-    /* This timeout is a temporary stop-gap until collecting stats is not bound to even-better
-     * and collecting stats is not interfered by logging configuration reloading
-     * Related to https://github.com/elastic/x-pack-kibana/issues/1301
-     */
     setTimeout(() => {
-      cleanup();
-      init();
+      stop();
+      start();
       this.log.info('Re-initialized Kibana Monitoring due to SIGHUP');
     }, 5 * 1000); // wait 5 seconds to avoid race condition with reloading logging configuration
   });
 
   return new Collector(server, {
     type: KIBANA_STATS_TYPE,
-    init,
-    fetch: buffer.flush,
-    fetchAfterInit: true,
```

Comment:
I intented to pull the removal of `fetchAfterInit` and `cleanup` as properties of the Collector constructor argument object, but getting that working without regressions of its own turned out to be a huge change that would have been temporary.